test/testlib/minikube_utilities.go

```diff
@@ -63,8 +65,32 @@ func AddTeardown(name string, teardownFunc func()) {
  * Nonetheless, there are use-cases where multiple Sleep() teardowns are useful - to allow inspecting different
  * intermediate states.
  */
-func AddDiagnosticTeardown(name string, teardownFunc func()) {
-	diagnosticTeardownLists[name] = append(diagnosticTeardownLists[name], teardownFunc)
+func AddDiagnosticTeardown(name string, condition interface{}, teardownFunc func()) {
+	tdfunc = func() {
+		shouldIdoIt := strings.EqualFold(AlwaysRunDiagnoseTeardowns, "true")
+
+		if !shouldIdoIt {
+			switch c := condition.(type) {
```

Comment:
This type matching stuff is pretty cool.

Reply:
Go has some really elegant syntax for things like that. :)